### PR TITLE
Adding a choice not to make batches in `timeseries_dataset_from_array`

### PR DESCRIPTION
**Pull request by europeanplaice**
I made a change in `timeseries_dataset_from_array` so that users can have a choice to make batches or not do.
It solves #15645.

### CODE REVIEW

File: keras/preprocessing/timeseries.py [hunk 1]
```diff
@@ -61,7 +61,7 @@ def timeseries_dataset_from_array(
       `data[i], data[i + r], ... data[i + sequence_length]`
       are used for create a sample sequence.
     batch_size: Number of timeseries samples in each batch
-      (except maybe the last one).
+      (except maybe the last one). If `None`, it doesn't make batches.
```
Comment (fchollet) [inline keras/preprocessing/timeseries.py:64]: ", the data will not be batched (the dataset will yield individual samples)."

**Review by fchollet — COMMENTED** [review]
Thanks for the PR! This is a reasonable feature.

Please:

- Also add this in `image_dataset_from_directory` and `text_dataset_from_directory`
- Add unit tests for the feature in all 3 functions

Thank you

**Comment by europeanplaice** [comment]
Thank you for your review. I added this in `image_dataset_from_directory` and `text_dataset_from_directory`.
Also, I added unit tests. 

The shuffle method depends on `batch_size`. If `batch_size` is None, I think `buffer_size` should be `len(dataset)` or something.

File: keras/preprocessing/timeseries.py (hunk 1 above)
Comment (europeanplaice) [reply keras/preprocessing/timeseries.py]: I changed as you suggested, Thank you!

**Comment by fchollet** [comment]
> should be len(dataset) or something.

In that case it should be a fixed constant, like 1024.

**Review by fchollet — APPROVED** [review]
LGTM, thank you for the contribution!

**Comment by europeanplaice** [comment]
My pleasure.

**Comment by fchollet** [comment]
There seems to be some failing tests -- please take a look.

**Pull request merged**